### PR TITLE
Use literal strings instead of regular expressions with filter_file 

### PR DESCRIPTION
**Pull request by gartung**
This fixes a problem with text relocation.

### CODE REVIEW

File: lib/spack/spack/relocate.py [hunk 1]
```diff
@@ -289,7 +289,7 @@ def relocate_text(path_names, old_dir, new_dir):
     """
     Replace old path with new path in text file path_name
     """
-    filter_file("r'%s'" % old_dir, "r'%s'" % new_dir,
```
Comment (scheibelp) [inline lib/spack/spack/relocate.py]: What was this originally intended to do? Did it break for certain inputs? If so, could you mention an example? To be clear I'm just looking for an answer here in the discussion thread (vs. updating the commit).

Comment (gartung) [reply lib/spack/spack/relocate.py]: The text is not getting changed during tarball extraction.

Comment (scheibelp) [reply lib/spack/spack/relocate.py]: Would you be able to add a unit test to confirm this function changes text as expected?

**Comment by gartung** [comment]
Submitting this commit with another branch that fixes tarball creation as well.

**Comment by gartung** [comment]
Unit tests don't allow me to install to a different spack install root.

**Comment by scheibelp** [comment]
> Unit tests don't allow me to install to a different spack install root.

I was looking for a test of the `relocate_text` function specifically, which doesn't require installing a spack package. You'd just have to create a file with contents that you expect should be relocated. Does that make sense?

File: lib/spack/spack/relocate.py (hunk 1 above)
Comment (gartung) [reply lib/spack/spack/relocate.py]: Moved to https://github.com/spack/spack/pull/6510

**Comment by gartung** [comment]
Need to resubmit for testing I guess.

**Pull request closed without merging**